spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/security/servlet/UserDetailsServiceAutoConfigurationTests.java

```diff
@@ -99,6 +102,19 @@ void defaultUserNotCreatedIfAuthenticationProviderBeanPresent(CapturedOutput out
 		});
 	}
 
+	@Test
+	void defaultUserNotCreatedIfResourceServerIsUsed() {
+		WebApplicationContextRunner webApplicationContextRunner = new WebApplicationContextRunner();
+		webApplicationContextRunner.withUserConfiguration(TestSecurityConfiguration.class)
+				.withConfiguration(AutoConfigurations.of(OAuth2ResourceServerAutoConfiguration.class))
+				.withPropertyValues(
+						"spring.security.oauth2.resourceserver.jwt.jwk-set-uri=https://example.com/oauth2/default/v1/keys")
+				.run((context) -> {
+					assertThat(context).hasSingleBean(JwtDecoder.class);
+					assertThat(context).doesNotHaveBean(UserDetailsService.class);
+				});
+	}
+
```

Comment:
@htztomic I think we need to add a test for `OAuth2TokenIntrospectionClient` too.